### PR TITLE
Prac 05 code review request

### PR DESCRIPTION
**Pull request by aidenlignieres**
Please check for any problems such as spelling, inconsistent naming, and logic.

@AdamD99

### CODE REVIEW

File: Prac05/emails.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        email_to_name[email] = name
+        email = input("Email: ")
+
+    for email, name in email_to_name.items():
+        print("{} ({})".format(name, email))
+
+
+def get_name_from_email(email):
+    prefix = email.split('@')[0]
+    parts = prefix.split('.')
```
Comment (AdamD99) [inline Prac05/emails.py:22]: prefix and parts could use the same variable name

File: Prac05/emails.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    name = " ".join(parts).title()
+    return name
```
Comment (AdamD99) [inline Prac05/emails.py:24]: can replace line 23 and 24 by returning the " ".join(parts).title() directly

File: Prac05/hex_colours.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+
+while colour_name != "":
+    # pass
+    if colour_name in CODE_TO_NAME:
+        print("The hex code for {} is {}".format(colour_name, CODE_TO_NAME[colour_name]))
+    else:
+        print("That colour is not in dictionaries.")
+        print("Please enter a valid colour")
+    colour_name = input("Colour name: ")
```
Comment (AdamD99) [inline Prac05/hex_colours.py:30]: easy to comprehend, good variable names

File: Prac05/state_names.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+
+CODE_TO_NAME = {"QLD": "Queensland", "NSW": "New South Wales", "NT": "Northern Territory", "WA": "Western Australia",
+                "ACT": "Australian Capital Territory", "VIC": "Victoria", "TAS": "Tasmania"}
+print(CODE_TO_NAME)
+
+state_code = input("Enter short state: ").upper()
+while state_code != "":
+    if state_code in CODE_TO_NAME:
+        print(state_code, "is", CODE_TO_NAME[state_code])
```
Comment (AdamD99) [inline Prac05/state_names.py:15]: use string formatting

File: Prac05/word_occurrences.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+line_of_text = input("Text: ")
+words = line_of_text.split()
+for word in words:
+    frequency = WORD_COUNT.get(word, 0)
+    WORD_COUNT[word] = frequency + 1
+
+words = list(WORD_COUNT.keys())
+words.sort()
+
+max_length = max(len(key) for key in WORD_COUNT)
```
Comment (AdamD99) [inline Prac05/word_occurrences.py:23]: variable name could be more descriptive, max length of what

**Review by AdamD99 — APPROVED** [review]

File: Prac05/emails.py (hunk 1 above)
Comment (lindsaymarkward) [reply Prac05/emails.py:22]: Why would reusing the same name for different things be a good idea? 🤔

File: Prac05/state_names.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    else:
+        print("Invalid short state")
+    state_code = input("Enter short state: ").upper()
+
+for key in CODE_TO_NAME:
```
Comment (lindsaymarkward) [inline Prac05/state_names.py:20]: Avoid using generic names like "key" when there are more meaningful ones.

**Pull request merged**